### PR TITLE
python3Packages.pyasn: init at 1.6.1

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
Packaging python module pyasn

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pyasn/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    inherit pname version;
+    sha256 = "sha256-6UK1SRY2Pse4tw6urs0OtOQe8bz0ojl7KabXFfzN+SU=";
+  };
+
+  doCheck = false; # Tests require internet connection which wont work
+
+  meta = with lib; {
+    description = "Offline IP address to Autonomous System Number lookup module";
+    homepage = "https://github.com/hadiasghari/pyasn";
+    license = "MIT";
```
Comment (davidak) [inline pkgs/development/python-modules/pyasn/default.nix:17]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/pyasn/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pname = "pyasn";
+  version = "1.6.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-6UK1SRY2Pse4tw6urs0OtOQe8bz0ojl7KabXFfzN+SU=";
+  };
+
+  doCheck = false; # Tests require internet connection which wont work
+
```
Comment (davidak) [inline pkgs/development/python-modules/pyasn/default.nix:13]:
```suggestion

pythonImportsCheck = [ "pyasn" ];

```

At least do a simple import test.

File: pkgs/development/python-modules/pyasn/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+buildPythonPackage rec {
+  pname = "pyasn";
+  version = "1.6.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-6UK1SRY2Pse4tw6urs0OtOQe8bz0ojl7KabXFfzN+SU=";
+  };
+
+  doCheck = false; # Tests require internet connection which wont work
```
Comment (davidak) [inline pkgs/development/python-modules/pyasn/default.nix:12]: not sure if we can and should allow it in such case

**Review by davidak — CHANGES_REQUESTED** [review]
Generally looks good

See comments

File: pkgs/development/python-modules/pyasn/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  meta = with lib; {
+    description = "Offline IP address to Autonomous System Number lookup module";
+    homepage = "https://github.com/hadiasghari/pyasn";
+    license = "MIT";
+    platforms = platforms.unix;  # arbitrary choice
```
Comment (fabaff) [inline pkgs/development/python-modules/pyasn/default.nix:18]: I don't think that we need this limitation. Common Python modules works on all platforms.

File: pkgs/development/python-modules/pyasn/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyasn/default.nix:17]: `radix.c` is "4-clause BSD license according to the file header.

```bash
license = with licenses; [ bsdOriginal /* and */ mit ];
```

**Review by fabaff — COMMENTED** [review]
Commit messages should be 

```bash
python3Pacakges.pyasn: init at 1.6.1
```

New packages need a maintainer.

**Comment by davidak** [comment]
>python3Pacakges.pyasn

should be `python3Packages.pyasn` (typo)

**Comment by onny** [comment]
> > python3Pacakges.pyasn
> 
> should be `python3Packages.pyasn` (typo)

Thanks its fixed :)

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 158130` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pyasn</li>
    <li>python39Packages.pyasn</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
Code looks good

Builds, works

Tested as described in README:

```
[davidak@gaming:~/code/nixpkgs]$ nixpkgs-review pr 158130
...

[nix-shell:~/.cache/nixpkgs-review/pr-158130]$ pyasn_util_download.py --latest
Connecting to ftp://archive.routeviews.org
Finding most recent archive in /bgpdata/2022.02/RIBS ...
Downloading ftp://archive.routeviews.org//bgpdata/2022.02/RIBS/rib.20220211.0600.bz2
 100%, 12830KB/s
Download complete.

[nix-shell:~/.cache/nixpkgs-review/pr-158130]$ pyasn_util_convert.py --single rib.20220211.0600.bz2 ipasn_20220211.0600.dat
Parsing MRT/RIB archive ..  MrtTD2Record (PEER-INDEX-TABLE, collector 2162111334, 78 peers)
  MRT record 100000 @12s
  MRT record 200000 @24s
  MRT record 300000 @36s
  MRT record 400000 @48s
  MRT record 500000 @60s
  WARNING: repeated prefix '168.232.84.0/23' maps to different origin (264562 vs 263562)
  WARNING: repeated prefix '177.73.70.0/24' maps to different origin (52698 vs 1)
  MRT record 600000 @72s
  MRT record 700000 @85s
  MRT record 800000 @98s
  MRT record 900000 @111s
IPASN database saved (931442 IPV4 + 0 IPV6 prefixes)

[nix-shell:~/.cache/nixpkgs-review/pr-158130]$ nix-shell -p 'python3.withPackages(ps: with ps; [ pyasn ])'
this derivation will be built:
  /nix/store/ksrkic220px1h4y7z1wgiym959vp0prs-python3-3.9.10-env.drv
building '/nix/store/ksrkic220px1h4y7z1wgiym959vp0prs-python3-3.9.10-env.drv'...
created 239 symlinks in user environment
created 239 symlinks in user environment

[nix-shell:~/.cache/nixpkgs-review/pr-158130]$ python
Python 3.9.10 (main, Jan 13 2022, 23:32:03) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyasn
>>> asndb = pyasn.pyasn('ipasn_20220211.0600.dat')
>>> asndb.lookup('8.8.8.8')
(15169, '8.8.8.0/24')
>>> asndb.get_as_prefixes(1128)
{'131.180.0.0/16', '130.161.0.0/16', '145.94.0.0/16'}
```

Thanks for the contribution! :sparkles:

**Pull request merged**